### PR TITLE
Locked files are now considered as access denied.

### PR DESCRIPTION
**Pull request by herve-er**

### CODE REVIEW

**Review by ChristopheLarchier — DISMISSED** [review]

File: src/libcommonserver/io/iohelper_win.cpp [hunk 1]
```diff
@@ -81,6 +81,8 @@ IoError ntStatus2ioError(NTSTATUS status) noexcept {
         case STATUS_SUCCESS:
             return IoError::Success;
         case STATUS_ACCESS_DENIED:
+        case ERROR_SHARING_VIOLATION:
+        case ERROR_LOCK_VIOLATION:
```
Comment (luc-guyot-infomaniak) [inline src/libcommonserver/io/iohelper_win.cpp:85]: Is there a simple unit test that can be written to prevent a regression?

Comment (herve-er) [reply src/libcommonserver/io/iohelper_win.cpp]: Added here: [0a15aaa](https://github.com/Infomaniak/desktop-kDrive/pull/381/commits/0a15aaa25e18c116700d456204e903293791741f)

Comment (herve-er) [reply src/libcommonserver/io/iohelper_win.cpp]: It look like this test isn't working on mac and linux.
I have to find a way to lock the file on all OS.

File: src/libcommonserver/io/iohelper_win.cpp [hunk 2]
```diff
@@ -583,7 +585,7 @@ static bool getRightsWindowsApi(const SyncPath &path, bool &read, bool &write, b
 
     // Get rights for trustee
     ACCESS_MASK rights = 0;
-    result = GetEffectiveRightsFromAcl(pfileACL, &IoHelper::getTrustee(), &rights);
+    //result = GetEffectiveRightsFromAcl(pfileACL, &IoHelper::getTrustee(), &rights);
```
Comment (ChristopheLarchier) [inline src/libcommonserver/io/iohelper_win.cpp:588]: Why is it commented now?

File: test/libcommonserver/io/testio.h [hunk 3]
```diff
@@ -43,7 +43,7 @@ struct IoHelperTests : public IoHelper {
 
 class TestIo : public CppUnit::TestFixture {
         CPPUNIT_TEST_SUITE(TestIo);
-        CPPUNIT_TEST(testCheckSetAndGetRights); // Keep this test before any tests that may use set/get rights functions
+       CPPUNIT_TEST(testCheckSetAndGetRights); // Keep this test before any tests that may use set/get rights functions
```
Comment (ChristopheLarchier) [inline test/libcommonserver/io/testio.h:46]: Indentation issue?

Comment (herve-er) [reply test/libcommonserver/io/testio.h]: Fixed here: [4ad40c1](https://github.com/Infomaniak/desktop-kDrive/pull/381/commits/4ad40c1a17386c7afbcab654548d81295a6aa1ca)

File: src/libcommonserver/io/iohelper_win.cpp (hunk 2 above)
Comment (herve-er) [reply src/libcommonserver/io/iohelper_win.cpp]: Fixed here: [4ad40c1](https://github.com/Infomaniak/desktop-kDrive/pull/381/commits/4ad40c1a17386c7afbcab654548d81295a6aa1ca)

File: src/libcommonserver/io/iohelper_win.cpp (hunk 1 above)
Comment (herve-er) [reply src/libcommonserver/io/iohelper_win.cpp]: Done

**Review by ChristopheLarchier — DISMISSED** [review]

File: test/libcommonserver/io/testio.cpp [hunk 4]
```diff
@@ -149,4 +149,27 @@ void TestIo::testLogDirectoryPath() {
...
+    file.close();
+
+    // Lock the file
+    auto hFile = CreateFile(lockedFile.c_str(), GENERIC_READ, 0, nullptr, OPEN_EXISTING, FILE_ATTRIBUTE_NORMAL, nullptr);
+    CPPUNIT_ASSERT(hFile != INVALID_HANDLE_VALUE);
+
+    // Try to delete the file
+    std::error_code ec;
+    std::filesystem::remove_all(lockedFile, ec);
+    IoError ioError = IoHelper::stdError2ioError(ec);
```
Comment (luc-guyot-infomaniak) [inline test/libcommonserver/io/testio.cpp:167]:
```suggestion
    const IoError ioError = IoHelper::stdError2ioError(ec);
```

**Review by luc-guyot-infomaniak — APPROVED** [review]

**Review by ClementKunz — APPROVED** [review]

**Pull request merged**